model-optimizer/extensions/middle/LayoutChangeForConstantShapePaths.py

```diff
@@ -100,17 +104,28 @@ def find_and_replace_pattern(self, graph: Graph):
         shape_sub_graph_end_points = self.find_shape_subgraph_endpoints([shape.out_port(0) for shape in shape_ops])
         for in_port in shape_sub_graph_end_points:
             name = in_port.node.soft_get('name', in_port.node.id)
+            shape = in_port.data.get_shape()
             rank = in_port.data.get_shape().item(0)
 
-            should_insert_gather = rank in [4, 5] and not len(in_port.node.soft_get('correct_out_data_layout', {}))
+            should_insert_gather = len(shape) == 1 and rank in [4, 5] and \
+                                   not len(in_port.node.soft_get('correct_out_data_layout', {}))
```

Comment:
Why don't you use `is_input_data_in_correct_layout` or `is_output_data_in_correct_layout` methods?